# Regional Microglial Transcriptomics and Epigenomics

Alexander V. Margetts1-3â€ , Samara J. Vilca1,2, Florence Bourgain-Guglielmetti1-3 & Luis M. Tuesta1-3*

1 Department of Psychiatry & Behavioral Sciences
2 Center for Therapeutic Innovation
3 Sylvester Comprehensive Cancer Center
  University of Miami Miller School of Medicine, Miami, FL 33136
* Corresponding author (ltuesta@miami.edu)
* Site Maintainer (alexmargetts@med.miami.edu)

Microglia, the innate immune cells in the central nervous system, exhibit distinct transcriptional profiles
across brain regions that are important for facilitating their specialized function. There has been recent
interest in identifying the epigenetic modifications associated with these distinct transcriptional profiles,
as these may improve our understanding of the underlying mechanisms governing the functional
specialization of microglia. One obstacle to achieving this goal is the large number of microglia required
to obtain a genome-wide profile for a single histone modification. Given the cellular and regional
heterogeneity of the brain, this would require pooling many samples which would impede biological
applications that are limited by numbers of available animals. To overcome this obstacle, we have
adapted a method of chromatin profiling known as Cleavage Under Targets and Tagmentation
(CUT&Tag-Direct) to profile histone modifications associated with regional differences in gene
expression throughout the brain reward system. Consistent with previous studies, we find that
transcriptional profiles of microglia vary by brain region. However, here we report that these regional
differences also exhibit transcriptional network signatures specific to each region. Additionally, we find
that these region-dependent network signatures are associated with differential deposition of H3K27ac
and H3K7me3, and while the H3K27me3 landscape is remarkably stable across brain regions, the
H3K27ac landscape is most consistent with the anatomical location of microglia which explain their
distinct transcriptional profiles. Altogether, these findings underscore the established role of H3K27me3
in cell fate determination and support the active role of H3K27ac in the dynamic regulation of microglial
gene expression. In this study, we report a molecular and computational framework that can be applied
to improve our understanding of the role of epigenetic regulation in microglia in both health and disease,
using as few as 2,500 cells per histone mark.

[Manuscript]

[Manuscript]: https://www.biorxiv.org/content/10.1101/2024.08.08.607229v1

[workflowr]: https://github.com/workflowr/workflowr
